spec/dummy/db/schema.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 ActiveRecord::Schema.define(version: 2019_02_216_153105) do
-  create_table :products, id: :serial do |t|
+  create_table :products do |t|
```

Comment:
what happens if we keep `id`?

Reply:
I get 
```
       ActiveRecord::NotNullViolation:
       SQLite3::ConstraintException: NOT NULL constraint failed: products.id
```

I have no real experience with sqlite but it seems like it does not set id autoincremental if ommited during creation.
As such I would need include the id in each product creation in the test-cases. 

so either ommit id: serial from schema or expricitly set id
```
          custom_product_class.create(
            configuration: Configuration.to_type.cast_value(attributes),
            id: 1
          )
 ```